fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -112,6 +155,70 @@ def sst_dst_to_dense(self, sst: Tensor, dst: Tensor = None) -> Tensor:
         """
         pass
 
+    def _transform(self, dense: Tensor) -> Tensor:
+        """
+        Applies the transformation algorithm chosen by the caller to the input dense tensor.
+        Returns the transformed tensor.
+
+        Args:
+            dense (Tensor):
+                Input dense tensor.
+        """
+        if self._algo is Algo.FFT:
+            transform = torch.fft.fft  # type: ignore
+        elif self._algo is Algo.DCT:
+            raise NotImplementedError
+        return transform(dense)
+
+    def _sparse_mask(
+        self,
+        in_tensor: Tensor,
+        top_k_element: Optional[int] = None,
+        top_k_percent: Optional[float] = None,
+        dim: int = -1,
+    ) -> Tensor:
+        """Returns a mask for a tensor with a certain sparsity level corresponding to either
+        the top_k_percent or the top_k_element of the tensor.
+
+        Args:
+            in_tensor (Tensor)
+                input torch tensor for which sparse mask is generated.
+            top_k_element (float, optional)
+                Number of top-k elements to retain in the dense tensor corresponding to the mask.
+            top_k_percent (float, optional)
+                Percent of top-k elements to retain in the dense tensor corresponding to the mask.
+            dim (int, optional):
+                The dimension on which the top-k is done..
+                E.g. -1 is the last dim. None means flatten and top-k on all dims.
+                There is no way to specify multiple dims other than None.
+        """
+        abs_tensor = torch.abs(in_tensor)
+
+        # when the criterion is top_k_percent
+        if self._dst_top_k_percent is not None or self._sst_top_k_percent is not None:
```

Comment:
it is very strange that you access those attributes from self within this function.

In fact, this function can be make into a helper that's outside of this class, right?

Does it really need any info from "self" class or it can operate independently?